From 2b515bbac0966f711a39bb33aa5497e31216cd51 Mon Sep 17 00:00:00 2001
From: Nipun Gupta <nipun.gupta@nxp.com>
Date: Tue, 20 Feb 2018 12:45:49 +0530
Subject: [PATCH 143/166] PVT: raw/dpaa2_cmdif: use iova-vaddr conversion
 macros

For I/O with hardware the addresses need to be converted
to iova from virtual address and vice-versa.

Signed-off-by: Nipun Gupta <nipun.gupta@nxp.com>
---
 drivers/raw/dpaa2_cmdif/dpaa2_cmdif.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/raw/dpaa2_cmdif/dpaa2_cmdif.c b/drivers/raw/dpaa2_cmdif/dpaa2_cmdif.c
index a166b91cb0ed..f718b52a80f1 100644
--- a/drivers/raw/dpaa2_cmdif/dpaa2_cmdif.c
+++ b/drivers/raw/dpaa2_cmdif/dpaa2_cmdif.c
@@ -95,7 +95,7 @@ dpaa2_cmdif_enqueue_bufs(struct rte_rawdev *dev,
 	fd.simple.bpid_offset = 0;
 	fd.simple.ctrl = 0;
 
-	DPAA2_SET_FD_ADDR(&fd, buffers[0]->buf_addr);
+	DPAA2_SET_FD_ADDR(&fd, DPAA2_VADDR_TO_IOVA(buffers[0]->buf_addr));
 	DPAA2_SET_FD_LEN(&fd, cmdif_send_cnxt->size);
 	DPAA2_SET_FD_FRC(&fd, cmdif_send_cnxt->frc);
 	DPAA2_SET_FD_FLC(&fd, cmdif_send_cnxt->flc);
@@ -149,7 +149,7 @@ dpaa2_cmdif_dequeue_bufs(struct rte_rawdev *dev,
 	qbman_pull_desc_set_fq(&pulldesc, rxq->fqid);
 	qbman_pull_desc_set_numframes(&pulldesc, 1);
 	qbman_pull_desc_set_storage(&pulldesc, dq_storage,
-		(dma_addr_t)dq_storage, 1);
+		(dma_addr_t)(DPAA2_VADDR_TO_IOVA(dq_storage)), 1);
 
 	while (1) {
 		if (qbman_swp_pull(swp, &pulldesc)) {
@@ -188,8 +188,8 @@ dpaa2_cmdif_dequeue_bufs(struct rte_rawdev *dev,
 
 	DPAA2_CMDIF_DEBUG("packet received\n");
 
-	buffers[0]->buf_addr = (void *)(DPAA2_GET_FD_ADDR(fd) +
-		DPAA2_GET_FD_OFFSET(fd));
+	buffers[0]->buf_addr = DPAA2_IOVA_TO_VADDR(DPAA2_GET_FD_ADDR(fd) +
+			DPAA2_GET_FD_OFFSET(fd));
 	cmdif_rcv_cnxt->size = DPAA2_GET_FD_LEN(fd);
 	cmdif_rcv_cnxt->flc = DPAA2_GET_FD_FLC(fd);
 	cmdif_rcv_cnxt->frc = DPAA2_GET_FD_FRC(fd);
-- 
2.14.3

